azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -90,15 +100,29 @@ def report_all_tracked():
                     CGroupsTelemetry._cgroup_metrics[key].marked_for_delete]:
             del CGroupsTelemetry._cgroup_metrics[key]
 
+        logger.reset_periodic()
+
         return collected_metrics
 
     @staticmethod
     def poll_all_tracked():
         with CGroupsTelemetry._rlock:
             for cgroup in CGroupsTelemetry._tracked[:]:
+
                 if cgroup.name not in CGroupsTelemetry._cgroup_metrics:
                     CGroupsTelemetry._cgroup_metrics[cgroup.name] = CgroupMetrics()
-                CGroupsTelemetry._cgroup_metrics[cgroup.name].add_new_data(cgroup.controller, cgroup.collect())
+
+                metric = None
+                # noinspection PyBroadException
+                try:
+                    metric = cgroup.collect()
+                except Exception as e:
+                    logger.periodic(logger.EVERY_HALF_HOUR,
```

Comment:
this should use the convenience functions (periodic_warn)

Reply:
The periodic_warn is a method of the Logger class, but we use the logger functions specified outside. This is why I pass the log level explicitly.